dex/networks/eth/common.go

```diff
@@ -0,0 +1,49 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+//go:build lgpl
+// +build lgpl
+
+package eth
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// DecodeCoinID decodes the coin ID into a common.Hash. For eth, there are no
+// funding coin IDs, just an account address. Care should be taken not to use
+// DecodeCoinID or (Driver).DecodeCoinID for account addresses.
+func DecodeCoinID(coinID []byte) (common.Hash, error) {
+	if len(coinID) != common.HashLength {
+		return common.Hash{}, fmt.Errorf("wrong coin ID length. wanted %d, got %d",
+			common.HashLength, len(coinID))
+	}
+	var h common.Hash
+	h.SetBytes(coinID)
+	return h, nil
+}
+
+// SecretHashSize is the byte-length of the hash of the secret key used in
+// swaps.
+const SecretHashSize = 32
+
+// ToGwei converts a *big.Int in wei (1e18 unit) to gwei (1e9 unit) as a uint64.
+// Errors if the amount of gwei is too big to fit fully into a uint64.
+func ToGwei(wei *big.Int) (uint64, error) {
+	gweiFactorBig := big.NewInt(GweiFactor)
+	gwei := big.NewInt(0).Div(wei, gweiFactorBig)
```

Comment:
Maybe not the best time to ask, but can you slip in a check that wei is not negative here? I believe negative will still fail below but with the wrong error message.